src/Workspaces/CSharp/Portable/CodeGeneration/ParameterGenerator.cs

```diff
@@ -139,21 +141,26 @@ internal static ParameterSyntax GetParameter(IParameterSymbol p, CodeGenerationO
             return ExpressionGenerator.GenerateExpression(parameter.Type, value, canUseFieldReference: true);
         }
 
-        private static SyntaxList<AttributeListSyntax> GenerateAttributes(
+        private static (SyntaxList<AttributeListSyntax>, NullableAnnotation) GenerateAttributes(
             IParameterSymbol parameter, bool isExplicit, CodeGenerationOptions options)
         {
+            var nullableAnnotation = parameter.NullableAnnotation;
+
             if (isExplicit)
             {
-                return default;
+                return (default, nullableAnnotation);
             }
 
             var attributes = parameter.GetAttributes();
+
+            AdjustNullableAnnotationByAttributes(parameter.Type, ref attributes, ref nullableAnnotation, isParameter: true);
```

Comment:
```suggestion
            var attributes = parameter.GetAttributes();
            var nullableAnnotation = parameter.NullableAnnotation;
            
            (attributes, nullableAnnotation) = AdjustNullableAnnotationByAttributes(parameter.Type, isParameter: true);
```

Reply:
@CyrusNajmabadi Do you mean to pass `attributes` and `nullableAnnotation` as by-value parameters and return the updated values as a tuple?

Reply:
yup!